### PR TITLE
Ampers - Maddie Shields

### PR DESCRIPTION
**Pull request by madaleines**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
How did you go about exploring the Edamam API, how did you try querying the API? | I used Postman with JSON to see the format to access the data
Describe your API Wrapper.  How did you decide on the methods you created? | I searched for the query and showed an individual recipe
Describe an edge case or failure case test you wrote for your API Wrapper. | If the word doesn't exist was one. If it does exist. If the query matches the search phrase
Explain how VCR aids in testing an API. | To help limit the amount of times I use my access tokens and also show the recorded versions that work.
What is the Heroku URL of your deployed application? | https://mvs-muncher.herokuapp.com/
Provide a link to the Trello board you used | https://trello.com/b/v5OKz7Pd/recipe-api-consumer

### CODE REVIEW

File: app/views/recipes/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<h1> Results for "<%= @query %>"</h1>
+<div class="">
+
+  <section class="recipes-list">
+    <% if @recipes %>
+    <ul>
+      <% @recipes.each do |recipe| %>
+      <li>
+        <%= image_tag(recipe.image) %>
```
Comment (CheezItMan) [inline app/views/recipes/index.html.erb:9]: Making the image a clickable link would improve the user experience.

File: app/views/recipes/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<h1> Results for "<%= @query %>"</h1>
+<div class="">
```
Comment (CheezItMan) [inline app/views/recipes/index.html.erb:2]: What purpose does this `div` element serve?

File: test/lib/edamam_api_wrapper_test.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    it "Can't search for a blank" do
+      VCR.use_cassette("recipes") do
+        response = EdamamApiWrapper.search_for("asfd")
+        response.must_equal []
+      end
+    end
+  end
+
+  describe "show_recipe" do
+    it "can show a single recipe" do
```
Comment (CheezItMan) [inline test/lib/edamam_api_wrapper_test.rb:30]: What about the negative case here?  What about if the URI is wrong?

File: test/test_helper.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+# To add Capybara feature tests add `gem "minitest-rails-capybara"`
+# to the test group in the Gemfile and uncomment the following:
+# require "minitest/rails/capybara"
+
+# Uncomment for awesome colorful output
+# require "minitest/pride"
+
+class ActiveSupport::TestCase
+  # Setup all fixtures in test/fixtures/*.yml for all tests in alphabetical order.
+  fixtures :all
```
Comment (CheezItMan) [inline test/test_helper.rb:28]: VCR configuration?

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Core Requirements** | 
Git hygiene | Reasonable number of commits and good commit messages
Comprehension questions | Check
**General** | 
Rails fundamentals (RESTful routing, use of named paths) | Check
Semantic HTML | Check, although the HTML is minimal.
Errors are reported to the user | If the URI is wrong the show action crashes.  There are no messages when there are no search results.  
API Wrapper to handle the API requests | Check
Controller testing | **MISSING**
Lib testing | Missing a negative case for the `show` action.  You also are missing the configuration for VCR in `test_helper.rb`
Search Functionality | Check
List Functionality |  Check
Show individual item functionality (link to original recipe opens in new tab) | Check, although the link to the original recipe is missing
**Styling** | 
Responsive layout | **Nope**
List View shows 10 items at a time/pagination | Check
The app is styled to create an attractive user interface | Minimal interface
**API Features** | 
The App attributes Edaman | **MISSING**
The VCR cassettes do not contain the API key | Without the configuration for VCR in test_helper.rb you aren't saving cassettes!  
**External Resources** | 
Link to Trello Board | Check, but I don't have access
Link to deployed app on Heroku | Check
**Overall** | You got it working, as noted you still need to work on testing and making sure VCR is configured.
